antarest/study/storage/variantstudy/model/command/generate_thermal_cluster_timeseries.py

```diff
@@ -0,0 +1,149 @@
+                return CommandOutput(status=True, message="All time series were generated successfully")
+
+    def _build_timeseries(self, study_data: FileStudy, tmp_path: Path) -> None:
+        # 1- Get the seed and nb_years to generate
+        # NB: Default seed in IHM Legacy: 5489, default seed in web: 3005489.
+        general_data = study_data.tree.get(["settings", "generaldata"], depth=3)
+        thermal_seed = general_data["seeds - Mersenne Twister"]["seed-tsgen-thermal"]
+        nb_years = general_data["general"]["nbtimeseriesthermal"]
+        # 2 - Build the generator
+        rng = MersenneTwisterRNG(seed=thermal_seed)
+        generator = ThermalDataGenerator(rng=rng, days=365)
+        # 3- Loop through areas in alphabetical order
+        areas: t.Dict[str, Area] = study_data.config.areas
+        sorted_areas = {k: areas[k] for k in sorted(areas)}
+        for area_id, area in sorted_areas.items():
+            # 4- Loop through thermal clusters in alphabetical order
+            sorted_thermals = sorted(area.thermals, key=lambda x: x.id)
+            for thermal in sorted_thermals:
+                # 5 - Filters out clusters with no generation
+                if thermal.gen_ts == LocalTSGenerationBehavior.FORCE_NO_GENERATION:
+                    continue
+                # 6- Build the cluster
+                modulation_matrix = study_data.tree.get(
+                    ["input", "thermal", "prepro", area_id, thermal.id.lower(), "modulation"]
+                )["data"]
+                modulation_capacity = np.array([row[2] for row in modulation_matrix])
+                ts_generator_matrix = study_data.tree.get(
+                    ["input", "thermal", "prepro", area_id, thermal.id.lower(), "data"]
+                )["data"]
+                fo_duration, po_duration, fo_rate, po_rate, npo_min, npo_max = [
+                    np.array(col, dtype=int) if i not in [2, 3] else np.array(col, dtype=float)
+                    for i, col in enumerate(zip(*ts_generator_matrix))
+                ]
+                cluster = ThermalCluster(
+                    unit_count=thermal.unit_count,
+                    nominal_power=thermal.nominal_capacity,
+                    modulation=modulation_capacity,
+                    fo_law=ProbabilityLaw(thermal.law_forced.value.upper()),
+                    fo_volatility=thermal.volatility_forced,
+                    po_law=ProbabilityLaw(thermal.law_planned.value.upper()),
+                    po_volatility=thermal.volatility_planned,
+                    fo_duration=fo_duration,
+                    fo_rate=fo_rate,
+                    po_duration=po_duration,
+                    po_rate=po_rate,
+                    npo_min=npo_min,
+                    npo_max=npo_max,
+                )
+                # 7- Generate the time-series
+                results = generator.generate_time_series(cluster, nb_years)
+                generated_matrix = results.available_power.T.tolist()
```

Comment:
For performances we'd better not call `tolist`, the create method seems to accept a numpy array.

Reply:
Indeed

Reply:
As numpy operator T returns a Fortran-contiguous array we have to copy it to make it C-contiguous. To avoid this however, we can change the code of the generation package to make it build the matrix the right way. 
I'll do a release of the package that does so.

Reply:
I did the release so I don't need the `tolist()` method anymore